### PR TITLE
Reconstruct full Capella block

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Extends the engine client with the ability to reconstruct full Capella blocks from blinded Capella blocks.

**Which issues(s) does this PR fix?**

Part of Capella

### CODE REVIEW

File: beacon-chain/blockchain/pow_block.go [hunk 1]
```diff
@@ -92,6 +93,7 @@ func (s *Service) getBlkParentHashAndTD(ctx context.Context, blkHash []byte) ([]
 	if blk == nil {
 		return nil, nil, errors.New("pow block is nil")
 	}
+	blk.Version = version.Bellatrix
```
Comment (nisdas) [inline beacon-chain/blockchain/pow_block.go:96]: Why set this ? Shouldn't it already have its version set in the `ExecutionBlockByHash` method ? Or are you not able to infer that

File: beacon-chain/execution/engine_client.go [hunk 2]
```diff
@@ -437,11 +437,12 @@ func (s *Service) ReconstructFullBellatrixBlock(
 	if executionBlock == nil {
 		return nil, fmt.Errorf("received nil execution block for request by hash %#x", executionBlockHash)
 	}
+	executionBlock.Version = blindedBlock.Version()
```
Comment (nisdas) [inline beacon-chain/execution/engine_client.go:440]: Same question here

File: encoding/bytesutil/bytes.go [hunk 3]
```diff
@@ -99,6 +99,15 @@ func ToBytes4(x []byte) [4]byte {
 	return y
 }
 
+// ToBytes20 is a convenience method for converting a byte slice to a fix
+// sized 20 byte array. This method will truncate the input if it is larger
+// than 20 bytes.
+func ToBytes20(x []byte) [20]byte {
```
Comment (nisdas) [inline encoding/bytesutil/bytes.go:105]: Can we add some unit tests for this ?

File: proto/engine/v1/json_marshal_unmarshal.go [hunk 4]
```diff
@@ -71,6 +91,26 @@ func (e *ExecutionBlock) UnmarshalJSON(enc []byte) error {
 	if !ok {
 		return errors.New("expected `totalDifficulty` field in JSON response")
 	}
+
+	rawWithdrawals, ok := decoded["withdrawals"]
+	if !ok || rawWithdrawals == nil {
+		e.Version = version.Bellatrix
```
Comment (nisdas) [inline proto/engine/v1/json_marshal_unmarshal.go:97]: Do we know this for sure ? This just means there are 0 withdrawals in the payload.

File: beacon-chain/blockchain/pow_block.go (hunk 1 above)
Comment (rkapka) [reply beacon-chain/blockchain/pow_block.go:96]: Not able to infer from how `ExecutionBlockByHash` is implemented. I can add a `version` argument, but this seems cleaner to me.

File: proto/engine/v1/json_marshal_unmarshal.go (hunk 4 above)
Comment (rkapka) [reply proto/engine/v1/json_marshal_unmarshal.go:97]: I asked on Discord and we have this now: https://github.com/ethereum/execution-apis/pull/334.

File: proto/engine/v1/json_marshal_unmarshal.go [hunk 5]
```diff
@@ -109,6 +147,82 @@ func (b *PayloadIDBytes) UnmarshalJSON(enc []byte) error {
...
+	if err != nil {
+		return nil, err
+	}
+	if err := w.UnmarshalJSON(b); err != nil {
+		return nil, err
+	}
+	return w, nil
+}
+
+func (w *Withdrawal) toWithdrawalJSON() (*withdrawalJSON, error) {
```
Comment (nisdas) [inline proto/engine/v1/json_marshal_unmarshal.go:169]: This is an odd method, why do we have it ?

Comment (rkapka) [reply proto/engine/v1/json_marshal_unmarshal.go:169]: I use it here:
```
ws := make([]*withdrawalJSON, len(e.Withdrawals))
for i, w := range e.Withdrawals {
	ws[i], err = w.toWithdrawalJSON()
	if err != nil {
		return nil, err
	}
}
decoded["withdrawals"] = ws
```

**Pull request merged**